### ISSUE TITLE
wast2json drops module with funcref table

### ISSUE THREAD

**Issue by alexcrichton** [WebAssembly/wabt#1291]
Given an input `*.wast` file:

```wat
(module)

(module
  (func $f)
  (func $g)

  (table $t funcref (elem (ref.func $f) (ref.null) (ref.func $g)))
)

(module)
```

Executing:

```sh
$ mkdir tmp
$ wast2json foo.wast -o tmp/foo.json --enable-all
$ ls tmp
foo.0.wasm  foo.1.wasm  foo.json
```

that lists only 2 modules but there should be 3, and looking at `foo.json` we see:

```json
{"source_filename": "foo.wast",
 "commands": [
  {"type": "module", "line": 1, "filename": "foo.0.wasm"}, 
  {"type": "module", "line": 9, "filename": "foo.1.wasm"}]}
```

which appears to drop the module in the middle?

This was discovered running https://github.com/WebAssembly/testsuite/blob/4e4ceb6ccbcd9ef39d464a9a6a61d94a7d57e7ba/proposals/bulk-memory-operations/elem.wast through `wast2json` and it seemed to silently drop [this module](https://github.com/WebAssembly/testsuite/blob/4e4ceb6ccbcd9ef39d464a9a6a61d94a7d57e7ba/proposals/bulk-memory-operations/elem.wast#L68-L73)

**Comment by binji** [comment]
Looks like there are two bugs here: 1) the parser is failing because the elem expression syntax is not currently supported when embedded directly in a table, and 2) there is missing error output. I'm working on a fix now.

**Comment by binji** [comment]
Fixed in #1293.
